### PR TITLE
feat(bot,rest,types): Add support for one-time entitlements

### LINKED ISSUE #3570: [api-docs] Premium Apps: One-Time Purchase SKUs + Managing Your Store (#6820)
A new commit was made into the api-docs repo: https://github.com/discord/discord-api-docs/commit/f0d1615432dada83e34c37972b3d1661735caf59
* Added docs for One-Time Purchase + Store

* Update docs/monetization/One-Time_Purchases.md

Co-authored-by: shay <swdewael@gmail.com>

* Update docs/Change_Log.md

Co-authored-by: shay <swdewael@gmail.com>

* Update docs/monetization/Managing_Your_Store.md

Co-authored-by: shay <swdewael@gmail.com>

* Update docs/monetization/One-Time_Purchases.md

Co-authored-by: shay <swdewael@gmail.com>

---------

Co-authored-by: shay <swdewael@gmail.com>

This is a bot created issue.

### PR DESCRIPTION
**Pull request by Fleny113**
Add support for one-time consumable and durable entitlements

Upstream: [discord/discord-api-docs#6820](https://github.com/discord/discord-api-docs/pull/6820)
fixes #3570

### CODE REVIEW

File: packages/bot/src/transformers/entitlement.ts
Comment (lts20050703) [inline packages/bot/src/transformers/entitlement.ts:17]:
```suggestion
  if (props.consumed && payload.consumed !== undefined) entitlement.consumed = payload.consumed
```
Actually... why do we even need to check if the payload contains something?

File: packages/rest/src/typings/routes.ts
Comment (lts20050703) [inline packages/rest/src/typings/routes.ts:271]:
```suggestion
    /** Route to consume an entitlement */
    consumeEntitlement: (applicationId: BigString, entitlementId: BigString) => string
```

File: packages/bot/src/transformers/entitlement.ts
Comment (Fleny113) [reply packages/bot/src/transformers/entitlement.ts:17]: idr exactly the reason but we do so...

Comment (lts20050703) [reply packages/bot/src/transformers/entitlement.ts]: I think I figured out the reason, sometimes we pass partial stuff to the transformers and abused `ts-expect-error` as well as pray that we handled optional payload properly in the transformer.

Comment (Fleny113) [reply packages/bot/src/transformers/entitlement.ts]: This is the reaction for #3613, i have opened it while doing stuff for #3618 and started thinking that the abuse of ts-expect-error is getting out of hand

Comment (lts20050703) [reply packages/bot/src/transformers/entitlement.ts]: Alright got it thank you! ❤️

**Review by lts20050703 — DISMISSED** [review]
LGTM!

**Review by AwesomeStickz — DISMISSED** [review]

**Review by AwesomeStickz — APPROVED** [review]

**Review by lts20050703 — APPROVED** [review]
GitHub can you stop dismissing my "stale" reviews when someone just merge `main` into PR?

Unless there's a merge conflict, is it really necessary to merge main into PR?

**Comment by AwesomeStickz** [comment]
> Unless there's a merge conflict, is it really necessary to merge main into PR?
I'd say so, because someone might not merge it properly as intended

**Pull request merged**
